src/main/java/com/direwolf20/buildinggadgets/common/items/pastes/ConstructionPasteContainer.java

```diff
@@ -0,0 +1,84 @@
+package com.direwolf20.buildinggadgets.common.items.pastes;
+
+import com.direwolf20.buildinggadgets.common.BuildingGadgets;
+import com.direwolf20.buildinggadgets.common.tools.InventoryManipulation;
+import com.direwolf20.buildinggadgets.common.tools.NBTTool;
+import net.minecraft.client.renderer.block.model.ModelBakery;
+import net.minecraft.client.renderer.block.model.ModelResourceLocation;
+import net.minecraft.client.util.ITooltipFlag;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.entity.player.InventoryPlayer;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.ActionResult;
+import net.minecraft.util.EnumActionResult;
+import net.minecraft.util.EnumHand;
+import net.minecraft.world.World;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+public class ConstructionPasteContainer extends GenericPasteContainer {
+
+    public int maxAmount;
```

Comment:
```suggestion
    private final IntSupplier maxAmount;
```
Use an Supplier based approach, and then call getAsInt in getMaxAmount.

It is critical to use an non 'static' (don't know how to express that otherwise in english) member here as Config values can change during GamePlay, or more importantly when joining a Server. Currently, this would completly ignore any reload/sync operation.

Also **DON'T USE PUBLIC NON_FINAL VARIABLES!!!**